components/brave_vpn/brave_vpn_connection_manager_mac.mm

```diff
@@ -0,0 +1,249 @@
+#include "base/strings/sys_string_conversions.h"
+
+// Referenced GuardianConnect implementation.
+// https://github.com/GuardianFirewall/GuardianConnect
+namespace brave_vpn {
+
+namespace {
+
+const NSString* kBraveVPNKey = @"BraveVPNKey";
+
+NSData* GetPasswordRefForAccount(const NSString* account_key) {
+  NSString* bundle_id = [[NSBundle mainBundle] bundleIdentifier];
+  CFTypeRef copy_result = NULL;
+  NSDictionary* query = @{
+    (__bridge id)kSecClass : (__bridge id)kSecClassGenericPassword,
+    (__bridge id)kSecAttrService : bundle_id,
+    (__bridge id)kSecAttrAccount : account_key,
+    (__bridge id)kSecMatchLimit : (__bridge id)kSecMatchLimitOne,
+    (__bridge id)kSecReturnPersistentRef : (__bridge id)kCFBooleanTrue,
+  };
+  OSStatus results = SecItemCopyMatching((__bridge CFDictionaryRef)query,
+                                         (CFTypeRef*)&copy_result);
+  if (results != errSecSuccess)
+    LOG(ERROR) << "Error: obtaining password ref(status:" << results << ")";
+  return (__bridge NSData*)copy_result;
+}
+
+OSStatus RemoveKeychanItemForAccount(const NSString* account_key) {
+  NSString* bundle_id = [[NSBundle mainBundle] bundleIdentifier];
+  NSDictionary* query = @{
+    (__bridge id)kSecClass : (__bridge id)kSecClassGenericPassword,
+    (__bridge id)kSecAttrService : bundle_id,
+    (__bridge id)kSecAttrAccount : account_key,
+    (__bridge id)kSecReturnPersistentRef : (__bridge id)kCFBooleanTrue,
+  };
+  OSStatus result = SecItemDelete((__bridge CFDictionaryRef)query);
+  if (result != errSecSuccess && result != errSecItemNotFound)
+    LOG(ERROR) << "Error: deleting password entry(status:" << result << ")";
+  return result;
+}
+
+OSStatus StorePassword(const NSString* password, const NSString* account_key) {
+  if (password == nil) {
+    LOG(ERROR) << "Error: password is empty";
+    return errSecParam;
+  }
+
+  CFTypeRef result = NULL;
+  NSString* bundle_id = [[NSBundle mainBundle] bundleIdentifier];
+  NSData* password_data = [password dataUsingEncoding:NSUTF8StringEncoding];
+  NSDictionary* sec_item = @{
+    (__bridge id)kSecClass : (__bridge id)kSecClassGenericPassword,
+    (__bridge id)kSecAttrService : bundle_id,
+    (__bridge id)kSecAttrSynchronizable : (__bridge id)kCFBooleanFalse,
+    (__bridge id)kSecAttrAccount : account_key,
+    (__bridge id)kSecValueData : password_data,
+  };
+  OSStatus status = SecItemAdd((__bridge CFDictionaryRef)sec_item, &result);
+  if (status != errSecSuccess) {
+    if (status == errSecDuplicateItem) {
+      VLOG(2) << "There is duplicated key in keychain. removing and re-adding.";
+      if (RemoveKeychanItemForAccount(account_key) == errSecSuccess)
+        return StorePassword(password, account_key);
+    }
+    LOG(ERROR) << "Error: storing password";
+  }
+
+  return status;
+}
+
+NEVPNProtocolIKEv2* CreateProtocolConfig(const BraveVPNConnectionInfo& info) {
+  NSString* hostname = [NSString stringWithUTF8String:info.hostname().c_str()];
```

Comment:
You could leave the code how you have it now, your you can use 
`NSString* hostname = base::SysUTF8ToNSString(info.hostname())`

Reply:
Fixed.